### PR TITLE
only calculate true bin once and save to propeller

### PR DESCRIPTION
**Pull request by ibsafa**

### CODE REVIEW

File: src/PROcreate.cxx
Comment (jacoblarkin) [inline src/PROcreate.cxx:1]: You should also fill this in process_CAFAna

File: src/PROsyst.cxx
Comment (jacoblarkin) [inline src/PROsyst.cxx:1]: You can delete the subchannel lines I think

**Review by jacoblarkin — CHANGES_REQUESTED** [review]
Just need to add the part to process_CAFAna. And if possible to delete, we shouldn't calculate subchannel if not needed

File: src/PROsyst.cxx [hunk 1]
```diff
@@ -312,7 +312,7 @@ namespace PROfit {
         PROspec ret(config.m_num_bins_total);
         for(size_t i = 0; i < prop.baseline.size(); ++i) {
             const int subchannel = FindSubchannelIndexFromGlobalBin(config, prop.bin_indices[i]);
```
Comment (jacoblarkin) [inline src/PROsyst.cxx:314]: This is unnecessary now

Comment (jacoblarkin) [reply src/PROsyst.cxx:314]: There are several of these in PROsyst

**Review by jacoblarkin — APPROVED** [review]
Thanks

**Review by markrosslonergan — COMMENTED** [review]
Looks good, 5% improvement here we come

**Pull request merged**